core-addon/FirefoxPrivilegedApi.js

```diff
@@ -32,8 +39,30 @@ this.firefoxPrivilegedApi = class extends ExtensionAPI {
           // eslint-disable-next-line no-undef
           let str = gUUIDGenerator.generateUUID().toString();
           return str.substring(1, str.length - 1);
-        }
-      }
-    }
+        },
+        async getRemoteSettings() {
+          return RemoteSettings(STUDY_COLLECTION_KEY).get();
```

Comment:
Can `RemoteSettings(STUDY_COLLECTION_KEY)` ever fail/return `undefined` or `null`?

Reply:
It should always return `[]` I believe, if there's any problem, but I'll double-check this.

Reply:
There is not global `try/finally` on the method, but yes, the returned value should always be a list ([source](https://searchfox.org/mozilla-central/rev/69babd862de70cabfa1d0a369d38e4881bd41e4d/services/settings/RemoteSettingsClient.jsm#343))